ophyd/areadetector/base.py

```diff
@@ -19,6 +20,77 @@ def __init__(self, prefix, **kwargs):
         super().__init__(prefix + '_RBV', write_pv=prefix, **kwargs)
 
 
+class NDDerivedSignal(DerivedSignal):
+    """
+    DerivedSignal to shape a flattened array
+
+    The purpose of this class is to take a flattened array and shape in its'
+    proper form. The shape of the final array may be static in which case the
+    shape and number of dimensions can be set as static integers. Otherwise,
+    other signals from this classes parent can inform what the proper shape of
+    the array
+    """
+    def __init__(self, derived_from, shape, num_dimensions,
+                 parent=None, **kwargs):
+        super().__init__(derived_from, parent=parent, **kwargs)
+        # Assemble our shape of signals
+        self._shape = []
+        for dim in shape:
```

Comment:
This looks like it only grabs these once at init.  If these change later I don't think it will be reflected here.  The ability to track the shape seems important for looking at ex, the output of the ROI plugin.

Reply:
If you pass in signals every call to `derived_shape` will be be updated, they are only static if you pass in integers. We will also be updated of any changes to the array shape via the `_array_shape_callback`

Reply:
Sorry, I was reading this as the train was pulling into Jamacia and miss-understood :disappointed: .